pkg/template/master_template.go

```diff
@@ -399,6 +399,19 @@ systemd:
       ExecStartPost=/usr/bin/systemctl restart k8s-kubelet.service
       [Install]
       WantedBy=multi-user.target
+  - name: ensure-kube-proxy-vpa.service
+    enabled: true
+    contents: |
+      [Unit]
+      Description=Enable VPA for kube-proxy
+      After=k8s-addons.service
+      [Service]
+      Type=oneshot
+      Restart=on-failure
+      ExecStartPre=/bin/bash -c "while ! /opt/bin/kubectl --kubeconfig=/etc/kubernetes/kubeconfig/addons.yaml get crd verticalpodautoscalers.autoscaling.k8s.io; do echo 'Waiting for VPA CRD to exists' && sleep 10; done"
+      ExecStart=/opt/bin/kubectl --kubeconfig=/etc/kubernetes/kubeconfig/addons.yaml apply -f /srv/kube-proxy-vpa.yaml
```

Comment:
This may be stupid from my side but, why wait until CRD exist instead of just applying the VPA CR until it succeeds? won't `kubectl` fails if trying to apply the VPA CR without the CRD being present?

The benefit I see is that if there is a problem with the API at that moment, it would retry to apply it, while on this approach it only tries to apply it once.

Reply:
It is basically the same.

```
Restart=on-failure
```

makes it so that if the `apply` fails it will retry a few times.
The real downside (in my opinion) of your approach is that the unit might run forever, while with mine it tries a few times and eventually gives up. Not a big deal either way. Happy to hear other opinions about this.